master/reference/cni-plugin/configuration.md

```diff
@@ -180,3 +200,70 @@ When using the CNI `host-local` IPAM plugin, a special value `usePodCidr` is all
 ```
 
 When making use of the `usePodCidr` option, the Calico CNI plugin requires read-only Kubernetes API access to the `Nodes` resource.
+
+##### IPAM Manipulation with Kubernetes Annotations
+
+###### Specifying IP Pools on a per-Pod basis
+
+In addition to specifying IP Pools in the CNI config as discussed above, Calico IPAM supports specifying IP Pools per-Pod using the following [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/).
+
+- `cni.projectcalico.org/ipv4pools`: A list of configured IPv4 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv4pools": "[\"192.168.0.0/16\"]"
+```
+
+- `cni.projectcalico.org/ipv6pools`: A list of configured IPv6 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv6pools": "[\"2001:db8::1/120\"]"
+```
+
+If provided, these IP Pools will override any IP Pools specified in the CNI config.
+
+
+
+  > **Note:**
+  >
+  > This requires the IP Pools to exist before `ipv4pools` or `ipv6pools` annotations are used.
+  > Requesting a subset of an IP Pool is not supported. IP Pools requested in the annotations must be the exact match to the IP Pools created.
+
+###### Requesting a Specific IP address
+
+You can also request a specific IP address through [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/) with Calico IPAM. 
+There are two annotations to request a specific IP address:
+
+- `cni.projectcalico.org/ipAddrs`: A list of IPv4 and/or IPv6 addresses to assign to the Pod. The requested IP addresses will be assigned from Calico IPAM and must exist within a configured IP Pool.
+
+  Example:
+
+```yaml
+annotations:
+        "cni.projectcalico.org/ipAddrs": "[\"192.168.0.1\"]"
+```
+
+- `cni.projectcalico.org/ipAddrsNoIpam`: A list of IPv4 and/or IPv6 addresses to assign to the Pod, bypassing IPAM. Any IP conflicts and routing have to be taken care of manually or by some other system.
+
+  Example: 
+
+```yaml
+annotations:
+        "cni.projectcalico.org/ipAddrsNoIpam": "[\"10.0.0.1\"]"
+```
+
+  > **Note:**
+  >
+  > - `ipAddrs` annotation requires the IP Pools to exist before the annotation is used.
```

Comment:
Missing "The" in a few places here.  

e.g "_The_ `ipAddrs` and`ipAddrsNoIpam` annotation..."


> - `ipAddrs` annotation requires the IP Pools to exist before the annotation is used.

We already say this above, and I think that is more clearly stated so lets just remove this bullet.

>  - `ipAddrsNoIpam` annotation should be used with caution. CNI plugin bypasses IPAM and simply assigns the requested IP to the pod when this annotation is used. It is up to the user to make sure there are no IP conflicts and IP is routable on the network. Calico will not advertise the IP address unless there is an IP Pool that contains this IP.

Same here.  I think this content would be better placed up above. See this comment above https://github.com/projectcalico/calico/pull/524/files#r103120620